src/PageCollector.ts

```diff
@@ -113,12 +138,54 @@ export class PageCollector<T> {
     this.#listeners.set(page, listeners);
   }
 
+  protected async subscribeForIssues(page: Page) {
+    if (this instanceof NetworkCollector) {
```

Comment:
Let's create a subclass for ConsoleCollector and invoke this by extending the addPage call. Let's also make sure we set up listeners first (by calling this.#initializePage) and then call `subscribeForIssues`